### PR TITLE
Don't try to enumerate tests until UIApplicationDidFinishLaunchingNotification

### PR DESCRIPTION
**Pull request by justinseanmartin**

### CODE REVIEW

**Comment by danielribeiro** [comment]
Looks good. Any comments @mikelikespie

File: TestDumper/TestDumper/Initialize.m [hunk 1]
```diff
@@ -118,50 +118,53 @@ static void PrintTestClass(FILE *outFile, NSString *testClass) {
...
 const int TEST_METHOD_LEVEL = 2;
 
+
 __attribute__((constructor))
 void initialize() {
     XCTestConfiguration *config = [[XCTestConfiguration alloc] init];
     NSString *testType = [NSString stringWithUTF8String: getenv("XCTEST_TYPE")];
     NSString *testTarget = [NSString stringWithUTF8String: getenv("XCTEST_TARGET")];
 
+    [[NSNotificationCenter defaultCenter] addObserverForName:UIApplicationDidFinishLaunchingNotification object:nil queue:nil usingBlock:^(NSNotification * _Nonnull note) {
```
Comment (danielribeiro) [inline TestDumper/TestDumper/Initialize.m]: It does work for logic tests.

**Comment by justinseanmartin** [comment]
I can't get it to pass the tests on my local machine, but that's apparently due to csrutil or something.

**Comment by justinseanmartin** [comment]
I just pushed a new commit where we only do this in the case of app tests and not logic tests, which seems safer. Please take another look.

**Comment by justinseanmartin** [comment]
@danielribeiro tested this and everything seems to be working properly

**Pull request merged**